### ISSUE TITLE
[Story] Add application settings screen for notifications

### ISSUE THREAD

**Issue by giomfo** [vector-im/element-meta#1676]
As a user, I want to be able to enable/disable my registration to the push notifications on my current device

Design: [iOS](https://www.figma.com/file/0MMNu7cTOzLOlWb7ctTkv3/Element-X?type=design&node-id=4498-170263&t=Q9TqyB73l0LtKsua-4), [Android](https://www.figma.com/file/0MMNu7cTOzLOlWb7ctTkv3/Element-X?type=design&node-id=5146-187003&t=Q9TqyB73l0LtKsua-4)

## Scope
```[tasklist]
### Android
- [ ] https://github.com/vector-im/element-x-android/issues/510
- [ ] https://github.com/vector-im/element-x-android/issues/531
- [ ] https://github.com/vector-im/element-x-android/issues/532
- [ ] https://github.com/vector-im/element-x-android/issues/533
- [ ] https://github.com/vector-im/element-x-android/issues/534
- [ ] https://github.com/vector-im/element-x-android/issues/536
- [ ] https://github.com/vector-im/element-x-android/issues/537
- [ ] https://github.com/vector-im/element-x-android/issues/538
- [ ] https://github.com/vector-im/element-x-android/issues/539
- [ ] https://github.com/vector-im/element-x-android/issues/540
```

```[tasklist]
### iOS
- [ ] https://github.com/vector-im/element-x-ios/issues/1022
- [ ] https://github.com/vector-im/element-x-ios/issues/1023
- [ ] https://github.com/vector-im/element-x-ios/issues/1024
- [ ] https://github.com/vector-im/element-x-ios/issues/1025
- [ ] https://github.com/vector-im/element-x-ios/issues/1026
- [ ] https://github.com/vector-im/element-x-ios/issues/1027
- [ ] https://github.com/vector-im/element-x-ios/issues/1028
- [ ] https://github.com/vector-im/element-x-ios/issues/1029
- [ ] https://github.com/vector-im/element-x-ios/issues/1030
- [ ] https://github.com/vector-im/element-x-ios/issues/1031
```

```[tasklist]
### Tasklist
- [ ] QA signoff on completion
- [ ] Design signoff on completion
- [ ] Product signoff on completion
```

##  Stretch goals
None at this time
<or add a tasklist>

## Out of scope
-

**Comment by giomfo** [comment]
@langleyd here is the update of this Story:
- first @amshakal and I decided to remove from the current scope the possibility to add a custom room notification setting from the global settings. This is doable from the room settings.

- iOS: all the work has been done (thx @nimau). It is available in the nightly release behind the Developer Option "Show notification settings" (disabled by default). All my tests were ok. This platform is ready for the QA/Design signoff.

- Android: This [PR](https://github.com/vector-im/element-x-android/pull/1037) is related to the 4 first Android issues (assigned to @yostyle). Only the UI has been implemented for this screen. You will need to plug this UI with the interface set up by @nimau at the Rust level.

cc @VolkerJunginger
